helios-tools/src/main/java/com/spotify/helios/cli/CliMain.java

```diff
@@ -67,17 +76,37 @@ public static void main(final String... args) {
 
   public CliMain(final PrintStream out, final PrintStream err, final String... args)
       throws Exception {
+    this.authProviderFactory = configureAuthProvider();
     this.parser = new CliParser(args);
     this.out = out;
     this.err = err;
     setupLogging();
   }
 
+  private static AuthProvider.Factory configureAuthProvider() {
```

Comment:
There was.

So, in an initial draft plugins had access to the CLI so that they could inject new parameters. In an effort to keep things smalled and focused, we dropped it for now. I still think it make sense though.